tests/bot/cogs/moderation/test_utils.py

```diff
@@ -0,0 +1,411 @@
+                    ), width=2048, placeholder="..."),
+                    colour=Colours.soft_red,
+                    url=utils.RULES_URL
+                ).set_author(
+                    name=utils.INFRACTION_AUTHOR_NAME,
+                    url=utils.RULES_URL,
+                    icon_url=Icons.defcon_denied
+                ).set_footer(text=utils.INFRACTION_APPEAL_FOOTER),
+                "send_result": True
+            }
+        ]
+
+        for case in test_cases:
+            args = case["args"]
+            expected = case["expected_output"]
+            send = case["send_result"]
+
+            with self.subTest(args=args, expected=expected, send=send):
+                send_private_embed_mock.reset_mock()
+
+                send_private_embed_mock.return_value = send
+                result = await utils.notify_infraction(*args)
+
+                self.assertEqual(send, result)
+
+                embed = send_private_embed_mock.call_args[0][1]
+
+                self.assertEqual(embed.to_dict(), expected.to_dict())
+
+                send_private_embed_mock.assert_awaited_once_with(args[0], embed)
+
+    @patch("bot.cogs.moderation.utils.send_private_embed")
+    async def test_notify_pardon(self, send_private_embed_mock):
+        """Should send an embed of a certain format as a DM and return `True` if DM successful."""
+        test_cases = [
+            {
+                "args": (self.user, "Test title", "Example content"),
+                "icon": Icons.user_verified,
+                "send_result": True
+            },
+            {
+                "args": (self.user, "Test title", "Example content", Icons.user_update),
+                "icon": Icons.user_update,
+                "send_result": False
+            }
+        ]
+
+        for case in test_cases:
+            args = case["args"]
+            send = case["send_result"]
+
+            expected = Embed(
+                description="Example content",
+                colour=Colours.soft_green
+            ).set_author(
+                name="Test title",
+                icon_url=case["icon"]
+            )
+
+            with self.subTest(args=args, expected=expected):
+                send_private_embed_mock.reset_mock()
+
+                send_private_embed_mock.return_value = send
+
+                result = await utils.notify_pardon(*args)
+                self.assertEqual(send, result)
+
+                embed = send_private_embed_mock.call_args[0][1]
+                self.assertEqual(embed.to_dict(), expected.to_dict())
+
+                send_private_embed_mock.assert_awaited_once_with(args[0], embed)
+
+    @patch("bot.cogs.moderation.utils.log")
+    async def test_post_user(self, log_mock):
+        """Should POST a new user and return the response if successful or otherwise send an error message."""
+        user = MockUser(discriminator=5678, id=1234, name="Test user")
+        some_mock = MagicMock(discriminator=3333)
```

Comment:
I'm not finding `some_mock` to be a particularly good name, if it's meant to represent something that is *not* a user, that's a useful piece of information that can be included in the variable name.

You're only setting the `discriminator` attr but later on you're accessing `id` and `name` too - is that by design?

Reply:
That was just for avoiding error. I added now id and name too.